### PR TITLE
Update mlpack

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

File: recipe/edit_python_cmake.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   add_definitions(-DARMA_EXTRA_DEBUG)
+diff -Naur a/src/mlpack/bindings/python/CMakeLists.txt b/src/mlpack/bindings/python/CMakeLists.txt
+--- a/src/mlpack/bindings/python/CMakeLists.txt	2018-11-13 20:33:55.000000000 -0200
++++ b/src/mlpack/bindings/python/CMakeLists.txt	2019-01-10 21:13:00.763993436 -0200
+@@ -169,7 +169,7 @@
+ install(CODE "execute_process(COMMAND mkdir -p $ENV{DESTDIR}${NEW_PYTHONPATH})")
+ install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE}
+     \"${CMAKE_BINARY_DIR}/src/mlpack/bindings/python/setup.py\" install
+-    --prefix=${CMAKE_INSTALL_PREFIX} --root=$ENV{DESTDIR}
++    --prefix=${CMAKE_INSTALL_PREFIX} --root=$ENV{DESTDIR} --single-version-externally-managed
```
Comment (rcurtin) [inline recipe/edit_python_cmake.patch:21]: If you like, we can incorporate this patch upstream, maybe via something like a CMake option called `PYTHON_SETUP_PY_EXTRA_PARAMETERS` or something?

Comment (marcelotrevisani) [reply recipe/edit_python_cmake.patch:21]: That would be good!

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
Closing in favor of https://github.com/conda-forge/mlpack-feedstock/pull/15

**Pull request closed without merging**